### DISCUSSION TITLE
👑 [需求] 比 LightFilter  更轻量的筛选组件

### DISCUSSION THREAD

**Discussion by guowenfh** [ant-design/pro-components#5666] [Ideas]
### 🔩 所属模块或组件

ProForm LightFilter

### 🥰 需求描述

目前的轻量筛选在结合表格使用时，已经蛮好用的了。但是在筛选项过多时，仍然会切分为多行占据屏幕空间。
我们可能需要比目前的轻量筛选更轻量的筛选组件，无论多少个筛选项都是在一个 input 下。动态添加筛选项，支持横向滚动，及下拉展示全部内容。

### ⛰ 功能需求适用场景

大量搜索项，可有机组合的情况。

### 🧐 解决方案

基于 目前的 LightFilter 进行增强。可以算是 LightFilter 增强版，可参考 https://devui.design/components/zh-cn/category-search/demo#auto-scroll

### 🚑 其他信息

参考截图：

![image](https://user-images.githubusercontent.com/15883471/180783798-a610bb8b-a269-4bc2-afe3-534dfc0186cb.png)
![image](https://user-images.githubusercontent.com/15883471/180784927-eb4283a8-8198-4e0f-a47f-c417756ec2dd.png)

**Comment by chenshuai2144** [comment]
这个是 tag 模式的select ，看起来很炫酷呀

**Comment by guowenfh** [comment]
不仅仅是 select 就像 目前的轻量筛选一样，每一项都能支持进行不同类型的数据的选择，包括 时间，单选，多选，输入框，tag 等等。但是对于用户而言，所展示的就一个输入框。

**Reply by chenshuai2144** [reply]
有详细设计稿吗？ 这个组件这么复杂交互估计很麻烦

**Reply by guowenfh** [reply]
设计稿没有，不过这个组件本身也是开源的，不过是 Angular 。https://devui.design/components/zh-cn/category-search/demo#auto-scroll 。在线可体验的

**Comment by allahbin-super** [comment]
这种筛选也是我目前所需要的一个组件

我现在用BetaSchemaForm做了个动态筛选，但是我发现BetaSchemaForm的自定义渲染存在问题

![image](https://user-images.githubusercontent.com/26215888/190904839-dc2f0a2c-6c34-436d-9ac6-4e8c1f2f980d.png)

当我使用renderFormItem来自定义渲染出来组件的时候，它被显示在一个气泡里面了

这样看着也很奇怪

![image](https://user-images.githubusercontent.com/26215888/190905010-d6ef7279-d4b6-4240-909b-0b59cfbf1076.png)

我觉得renderFormItem应该能够覆盖掉默认的那下拉气泡
